setup.py

```diff
@@ -88,5 +88,6 @@ def _restore_install_lib(self):
 
 if __name__ == '__main__':
     # allow setup.py to run from another directory
-    here and os.chdir(here)
+    if here:
+        os.chdir(here)
```

Comment:
Again, let's avoid changing the semantics when adding type checking, unless there's a legitimate error, and even then, better to correct it in a separate contribution.

Reply:
Before I open a PR, this raises the following in pyright and mypy:
- Expression value is unused Pylance([reportUnusedExpression](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportUnusedExpression))
- "chdir" does not return a value Mypy([func-returns-value](https://mypy.readthedocs.io/en/latest/_refs.html#code-func-returns-value))

In other words, `os.chdir` has no return value, and that value is being used in a condition.

I understand the fluent/functional type-guard style here. I use it a lot in TypeScript and C#. Not so common in Python in my experience (I guess it's not "pythonic"?).
This is the only place in setuptool's codebase I found so far using it (at least in the code type-checked by this PR, which isn't as that much) But if you wanna allow that style, I can disable the appropriate rules in the type-checkers. Or at least ignore it for this one instance if you absolutely want to keep like as-is.